10up/block-components
# Add ability to show inline media controls for the `Image` component

### Description of the Change

Sometimes, we have blocks that require multiple images, and this change aims at offering an inline solution for replacing and or removing images where the toolbar replaces flow isn't ideal. 


Closes https://github.com/10up/block-components/issues/164

### How to test the Change

#### Steps to test
- Add the _Multiple Image Example_ block to the page.  
<img width="219" alt="image" src="https://user-images.githubusercontent.com/3490023/201422748-679f923c-2308-47b9-b964-e63d8a2d3011.png">

- You should see media placeholders for 3 images.  
<img width="350" alt="image" src="https://user-images.githubusercontent.com/3490023/201422841-c0fd9d89-3c8a-4143-89b1-2e25e487ac4d.png">

- Once you have set an image, you should see the "Replace" and "Remove" icon buttons when hovering over the image. This is because the new (optional) `hasInlineControls` props have been set to `true`.  
<img width="350" alt="image" src="https://user-images.githubusercontent.com/3490023/201421686-4deabe4e-f3d5-48c0-838c-26b52bec1855.png">  

**_Note_**: If text is preferred over icons, this could be easily replaced. We should probably get UX/Design feedback on the icons. 

- You should be able to replace the image with another and see the updated image display (updating the image id in the DB)
- The "Remove" icon button should not be visible if the `isOptional` props have been set to `false`.  
<img width="350" alt="image" src="https://user-images.githubusercontent.com/3490023/201427859-57306538-7a48-47fa-a07d-56a95f64a12a.png">


#### Usage example:
##### Optional
```
<Image 
  id={imageId} 
  size="large" 
  onSelect={(image) => setAttributes({imageId: image.id })} 
  className="example-image" 
  focalPoint={focalPoint} 
  onChangeFocalPoint={(value) => setAttributes({focalPoint: value})} 
  onRemove={() => setAttributes({imageId: null})} 
  hasInlineControls={true} 
/>
```
##### Required
```
<Image 
  id={imageId} 
  size="large" 
  onSelect={(image) => setAttributes({imageId: image.id })} 
  className="example-image" 
  focalPoint={focalPoint} 
  onChangeFocalPoint={(value) => setAttributes({focalPoint: value})} 
  onRemove={() => setAttributes({imageId: null})}
  hasInlineControls={true} 
  isOptional={false}
/>
```

### Changelog Entry

> Added - Ability to show inline media controls for the `Image` component


### Credits

Props @fabiankaegy 


### Checklist:
